plugins/upower.c

```diff
@@ -0,0 +1,276 @@
+/*
+ *
+ *  oFono - Open Source Telephony
+ *
+ *  Copyright (C) 2015 Canonical Ltd.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <glib.h>
+#include <gdbus.h>
+#include <ofono.h>
+
+#define OFONO_API_SUBJECT_TO_CHANGE
+#include <ofono/plugin.h>
+#include <ofono/modem.h>
+#include <ofono/log.h>
+#include <ofono/emulator.h>
+
+#define UPOWER_SERVICE            "org.freedesktop.UPower"
+#define UPOWER_DEVICE_INTERFACE   UPOWER_SERVICE ".Device"
+#define UPOWER_BASE_BATT_PATH     "/org/freedesktop/UPower/devices/battery_"
+#define UPOWER_TOUCH_BATT_PATH    UPOWER_BASE_BATT_PATH "battery"
```

Comment:
Can't we use the 'Type' property of the org.freedesktop.UPower.Device interface? See http://upower.freedesktop.org/docs/Device.html#Device:Type

If we use that in conjunction with IsPresent we should cover most of the cases. The case where a device has two batteries should be really rare ..

That would make this independent of what we have on Ubuntu Touch.